### PR TITLE
Expand alias support in CryptoCreate & CryptoTransfer Transactions

### PR DESCRIPTION
**Pull request by Nana-EC**
Signed-off-by: Nana Essilfie-Conduah <nana@swirldslabs.com>

**Description**:
Expand alias support in CryptoCreate & CryptoTransfer Transactions

This HIP proposes an expansion to the ways in which an account with an alias can be created and referenced to ensure compatibility with users coming from existing chains with `ECDSA` keys and provide greater applicability to smart contract scenarios.

It is thus proposed to

- Add an alias to public key mapping validation check to ensure single user ownership
- Expand `CryptoCreate` transaction to support setting an alias using a public-key
- Expand `CryptoTransfer` and `CryptoCreate` transactions to support the input of an alias of the Ethereum account address style with no public key

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: HIP/hip-0000-expand-alias-crypto-create-transfer.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+Add alias bytes property to the `CryptoCreateTransactionBody` [proto file](https://github.com/hashgraph/hedera-protobufs/blob/main/services/crypto_create.proto)
+
+```proto
+message CryptoCreateTransactionBody {
+	...
+	
+	/**
+   * The bytes to be used as the account's alias. The bytes will be 1 of 2 options. It will be the serialization
+   * of a protobuf Key message for an ED25519/ECDSA primitive key type. If the account is ECDSA based it may also be the public-address, calcluated as the last 20 bytes of the keccak-256 of the ECDSA primitive key.
```
Comment (MiroslavGatsanoga) [inline HIP/hip-0000-expand-alias-crypto-create-transfer.md:123]: Are we going to support ED25519 key aliases for CryptoCreate? See related discussion: https://github.com/hashgraph/hedera-services/pull/3943#discussion_r972471310.

File: HIP/hip-0000-expand-alias-crypto-create-transfer.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+- Accept the use of an alias during account creation
+- Add support to check if an account exists with a given alias
+
+### Performance Implications
+
+The mass adoption of alias setting on new and existing accounts will have minor but notable memory implications on the Hedera ecosystem
+
+`CryptoCreate` transactions will see extra bytes for every account created with an alias
+
+The size of the bytes in each case depends on the governing account key type. For an `ECDSA` based account the alias will always be a valid Ethereum address and will be 20 bytes long. In the `ED25519` case the alias will be a valid protobuf serialized public key of 35 or 68 bytes depending on encoding.
```
Comment (MiroslavGatsanoga) [inline HIP/hip-0000-expand-alias-crypto-create-transfer.md:186]: Are we going to support ED25519 key aliases for CryptoCreate? See related discussion: https://github.com/hashgraph/hedera-services/pull/3943#discussion_r972471310.

File: HIP/hip-0000-expand-alias-crypto-create-transfer.md (hunk 1 above)
Comment (Nana-EC) [reply HIP/hip-0000-expand-alias-crypto-create-transfer.md:123]: Yes. Aliases will remain both simple ED or EC keys

**Review by mgarbs — APPROVED** [review]
LGTM

**Review by mgarbs — APPROVED** [review]
LGTM

**Pull request merged**